topic-operator fails to use truststore/keystore as specified with env vars

Running strimzi/topic-operator:0.8.2 standalone on Oracle Cloud Infrastructure/Oracle Kubernetes Engine.

We have added an initcontainer to the manifests provided with examples/install/topic-operator.  We successfully construct a truststore, a keystore, & passwords.  When these are passed in to STO as environment vars, topic-operator code ignores them and attempts to build keystore & truststore in /tmp/topic-operator, as indicated in the attached log file.

org.apache.kafka.common.KafkaException: Failed create new KafkaAdminClient
Caused by: org.apache.kafka.common.KafkaException: org.apache.kafka.common.KafkaException: org.apache.kafka.common.KafkaException: Failed to load SSL keystore /tmp/topic-operator/replication.keystore.p12 of type JKS

We log in to the STO container and verify the following:
\# env |grep STRIMZI
STRIMZI_ZOOKEEPER_SESSION_TIMEOUT_MS=20000
STRIMZI_CONFIGMAP_LABELS=strimzi.io/kind=topic
STRIMZI_TLS_ENABLED=true
STRIMZI_NAMESPACE=kweet-psr
STRIMZI_LOG_LEVEL=DEBUG
**STRIMZI_TRUSTSTORE_PASSWORD=GxeGs2Denj**
STRIMZI_VERSION=0.8.2
**STRIMZI_KEYSTORE_PASSWORD=GxeGs2Denj**
**STRIMZI_TRUSTSTORE_LOCATION=/jks/kafka.client.truststore.jks**
**STRIMZI_KEYSTORE_LOCATION=/jks/kafka.client.keystore.jks**
STRIMZI_ZOOKEEPER_CONNECT=10.0.6.38:2181
STRIMZI_TOPIC_METADATA_MAX_ATTEMPTS=6
STRIMZI_KAFKA_BOOTSTRAP_SERVERS=10.0.6.39:9092
STRIMZI_FULL_RECONCILIATION_INTERVAL_MS=900000 

NOTE: The logfile indicates topic-operator is using different values for STRIMZI_TRUSTSTORE_LOCATION AND STRIMZI_KEYSTORE_LOCATION.

Check the jks files we constructed.
\# ls -l /jks
total 8
-rw-r--r--. 1 root root 2328 Nov  2 19:38 kafka.client.keystore.jks
-rw-r--r--. 1 root root 1073 Nov  2 19:38 kafka.client.truststore.jks
(Present and accounted for.)

Check the jks files the topic-operator tried to build.
\
[STO-tls-failure-2018-11-02--19-39-13.log](https://github.com/strimzi/strimzi-kafka-operator/files/2543952/STO-tls-failure-2018-11-02--19-39-13.log)
\# ls -l /tmp/topic-operator/
total 0
( None there.  )

Please advise.

--- a/topic-operator/scripts/topic_operator_run.sh
+++ b/topic-operator/scripts/topic_operator_run.sh
@@ -6,21 +6,21 @@ then
 fi
 
 if [ "$STRIMZI_TLS_ENABLED" = "true" ]; then
+    if [ -z "$STRIMZI_TRUSTSTORE_LOCATION" ] && [ -z "$STRIMZI_KEYSTORE_LOCATION" ]; then
+        # Generate temporary keystore password
+        export CERTS_STORE_PASSWORD=$(< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c32)
 
-    # Generate temporary keystore password
-    export CERTS_STORE_PASSWORD=$(< /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c32)
+        mkdir -p /tmp/topic-operator
 
-    mkdir -p /tmp/topic-operator
+        # Import certificates into keystore and truststore
+        /bin/tls_prepare_certificates.sh
 
-    # Import certificates into keystore and truststore
-    /bin/tls_prepare_certificates.sh
+        export STRIMZI_TRUSTSTORE_LOCATION=/tmp/topic-operator/replication.truststore.p12
+        export STRIMZI_TRUSTSTORE_PASSWORD=$CERTS_STORE_PASSWORD
 
-    export STRIMZI_TRUSTSTORE_LOCATION=/tmp/topic-operator/replication.truststore.p12
-    export STRIMZI_TRUSTSTORE_PASSWORD=$CERTS_STORE_PASSWORD
-
-    export STRIMZI_KEYSTORE_LOCATION=/tmp/topic-operator/replication.keystore.p12
-    export STRIMZI_KEYSTORE_PASSWORD=$CERTS_STORE_PASSWORD
-
+        export STRIMZI_KEYSTORE_LOCATION=/tmp/topic-operator/replication.keystore.p12
+        export STRIMZI_KEYSTORE_PASSWORD=$CERTS_STORE_PASSWORD
+    fi
 fi
 
 exec /bin/launch_java.sh $1

